scripts/prod/build_prod.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+# Make sure the script can only be run via SocialPredict Script
+[ -z "$CALLED_FROM_SOCIALPREDICT" ] && { echo "Not called from SocialPredict"; exit 42; }
+
+# Get the list of current docker images and convert to array
+IMAGES=$( docker images -a | awk '{print $1}' )
+IMAGES_ARRAY=($IMAGES// / })
+
+# Function to replace API_URI in frontend/src/config.js
+frontend_api_uri() {
+	template="$SCRIPT_DIR/frontend/src/config.js.template"
+	file="$SCRIPT_DIR/frontend/src/config.js"
+	export API_DOMAIN="'https://${DOMAIN}'"
+	envsubst < $template > $file
+}
+
+# Function to build frontend image
+build_frontend() {
+        echo "### Building Frontend Image ..."
+
+	# Update API_URI
+	frontend_api_uri
+
+        # Get frontend directory
+        FRONTEND_DIR="$( readlink -f "$SCRIPT_DIR/frontend" )"
+
+        # Build Image
+        docker build --no-cache -t "${FRONTEND_IMAGE_NAME}" -f "$FRONTEND_DIR/Dockerfile" "$FRONTEND_DIR/."
+        echo "Frontend Image Built"
+}
+
+# Function to build backend image
+build_backend() {
+        echo "### Building Backend Image ..."
+
+        # Get backend directory
+        BACKEND_DIR="$( readlink -f "$SCRIPT_DIR/backend" )"
+
+        # Build Image
+        docker build --no-cache -t "${BACKEND_IMAGE_NAME}" -f "$BACKEND_DIR/Dockerfile" "$BACKEND_DIR/."
+        echo "Backend Image Built"
+}
+
+# Search for images
+echo "### Searching for Docker Images ..."
+sleep 1;
+
+# Check if backend image exists
+echo "Searching for Backend Image ..."
+sleep 1;
+if [[ ${IMAGES_ARRAY[@]} =~ "${BACKEND_IMAGE_NAME}" ]]; then
+        read -p "Backend Image Found. Do you want to re-build it? (y/N) " DECISION
+
+	if [ "$DECISION" = "Y" ] || [ "$DECISION" = "y" ]; then
+		echo "Deleting Image..."
+		docker rmi "${BACKEND_IMAGE_NAME}"
+		echo "Image Deleted."
+		echo "Re-building image."
+		build_backend
+	fi
+        sleep 1;
+else
+        echo "Backend Image Not Found."
+        echo "Building now."
+        sleep 1;
+        build_backend
+fi
+
+# Check if frontend image exists
+echo
+echo "Searching for Frontend Image ..."
+if [[ ${IMAGES_ARRAY[@]} =~ "${FRONTEND_IMAGE_NAME}" ]]; then
+        read -p "Frontend Image Found. Do you want to re-build it? (y/N) " DECISION
+
+        if [ "$DECISION" = "Y" ] || [ "$DECISION" = "y" ]; then
+                echo "Deleting Image..."
+                docker rmi "${FRONTEND_IMAGE_NAME}"
+                echo "Image Deleted."
+                echo "Re-building image."
+                build_frontend
+        fi
+        sleep 1;
+
+else
+        echo "Frontend Image Not Found."
+        echo "Building now."
+        sleep 1;
+        build_frontend
+fi
+
+echo
+sleep 1;
+
+# Pull remaining images
+echo "Pulling remaining images ..."
```

Comment:
Got an error here. Trace...

```
root@breirfoxforecast-alpha:/home/testdeploy/socialpredict# nslookup -type=A www.brierfoxforecast.com
Server:		127.0.0.53
Address:	127.0.0.53#53

Non-authoritative answer:
Name:	www.brierfoxforecast.com
Address: 143.198.177.112

root@breirfoxforecast-alpha:/home/testdeploy/socialpredict# ./SocialPredict install
### Building and Deploying SocialPredict ...

### Checking that docker compose is installed ...
Found docker compose.

### Select Application Envrionment:
1) Development
2) Production
3) Quit
Please enter your choice: 2

Building for Production

### First time running the script ...
Let's initialize the application ...
What domain do you wish to use for the application? www.brierfoxforecast.com
Setting DOMAIN to: www.brierfoxforecast.com

What email address do you wish to use for the SSL Certificate? StarlingsOne@proton.me
Setting EMAIL to: StarlingsOne@proton.me

Specify a username for the Database User (default is insecure and not allowed): j711fMI1KEXh9GJA
Username must be at least 5 characters long and include special characters.
Specify a username for the Database User (default is insecure and not allowed): j711fMI1KEXh&
Setting Database User to: j711fMI1KEXh&

Specify password for the Database User (default is insecure and not allowed): j711fMI1KEXh%
Setting Database Password to: j711fMI1KEXh%

Specify the name for the Database (default: socialpredict_db):
Setting Database Name to: socialpredict_db

Specify the password for the Admin User (default is insecure and not allowed): 711fMI1KEXh%
Setting Admin Password to: 711fMI1KEXh%
Application initialized successfully.

### Searching for Docker Images ...
Searching for Backend Image ...
Backend Image Found. Do you want to re-build it? (y/N) n

Searching for Frontend Image ...
Frontend Image Found. Do you want to re-build it? (y/N) n

Pulling remaining images ...
failed to read /home/testdeploy/socialpredict/.env: line 15: unexpected character "'" in variable name
```

Reply:
Cannot replicate. Remove .env file and try again.

Reply:
Line 15 was the postgres user so the error was likely due to requiring a special character in the postgres username. Your recent fix from earlier today likely eliminated this. Resolving.